### PR TITLE
SI-5779: Wrong warning message (comparing Number)

### PR DESCRIPTION
**Pull request by som-snytt**
Probably this bug won't exist in some future version of Scala.  

It's hardly a bug; more like a gnat.

BoxesRuntime knows how to compare java.lang.Number, so we must not warn.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/RefChecks.scala [hunk 1]
```diff
@@ -1123,10 +1126,11 @@ abstract class RefChecks extends InfoTransform with reflect.internal.transform.R
           else if (!isUnit(actual) && !isMaybeValue(actual))  // () == "abc"
             nonSensiblyNeq()
         }
-        else if (isNumeric(receiver)) {
-          if (!isNumeric(actual) && !forMSIL)
+        else if (isNumeric(receiver) || (!forMSIL && isJavaNumeric(receiver))) {
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/RefChecks.scala]: these conditions are getting pretty unwieldy -- could you please simplify or elucidate with a comment?
think of the poor reviewer and future-selves reading this code

**Comment by adriaanm** [comment]
thanks for the fix! it looks like something that should go in but we have limited reviewing bandwidth
could you please help us out juuuust a bit more and make it obviously correct (either by introducing variables that name the conditions, comments (including a comment here?))

**Comment by som-snytt** [comment]
OK.  We're sensitive to the bandwidth issue.  At a certain point, a low-priority or no-priority pull req is just inefficient or a waste of time.

I'll comment.  (It begs for refactor that is out of scope.  No doubt Paul will do that while on sabbatical.)  (I picked this bug because I just fixed a bug in its neighborhood.)

**Comment by som-snytt** [comment]
The change amounts to: add isJavaNumber to isAnyNumber (as commented).

This commit isn't squashed; the diff would be cleaner.

**Comment by adriaanm** [comment]
LGTM -- thanks! waiting for the build bot to churn through the second commit and I'll merge (unless someone beats me to it, that'd be okay too)

**Comment by adriaanm** [comment]
build ok: https://scala-webapps.epfl.ch/jenkins/view/pr-validators/job/pr-scala-testsuite-linux-opt/152/

**Pull request merged**